src/collection-preferences/content-display/__tests__/content-display.test.tsx

```diff
@@ -0,0 +1,398 @@
+      const titleId = wrapper.findTitle().getElement().id;
+      expect(titleId).toBeTruthy();
+      const descriptionId = wrapper.findDescription().getElement().id;
+      expect(descriptionId).toBeTruthy();
+      expect(titleId).not.toBe(descriptionId);
+      const list = wrapper.findAll('UL')[0].getElement();
+      expect(list.getAttribute('aria-labelledby')).toBe(titleId);
+      expect(list.getAttribute('aria-describedby')).toBe(descriptionId);
+    });
+
+    it('displays list of options with correct semantics', () => {
+      const wrapper = renderContentDisplay();
+      const options = wrapper.findOptions();
+      for (let i = 0; i < options.length; i++) {
+        testOption({ wrapper, option: options[i], index: i });
+      }
+    });
+  });
+
+  describe('Content visibility', () => {
+    it('toggles content visibility', () => {
+      const collectionPreferencesWrapper = renderCollectionPreferences({
+        contentDisplayPreference,
+        onConfirm: () => null,
+        preferences: {
+          contentDisplay: [
+            {
+              id: 'id1',
+              visible: true,
+            },
+            {
+              id: 'id2',
+              visible: false,
+            },
+            {
+              id: 'id3',
+              visible: true,
+            },
+            { id: 'id4', visible: true },
+          ],
+        },
+      });
+      collectionPreferencesWrapper.findTriggerButton().click();
+      const contentDisplayPreferenceWrapper = collectionPreferencesWrapper.findModal()!.findContentDisplayPreference()!;
+      expectVisibilityStatus({ wrapper: contentDisplayPreferenceWrapper, statuses: [true, false, true, true] });
+      contentDisplayPreferenceWrapper.findOptions()[1].findToggle().findNativeInput().click();
+      expectVisibilityStatus({ wrapper: contentDisplayPreferenceWrapper, statuses: [true, true, true, true] });
+      contentDisplayPreferenceWrapper.findOptions()[2].findToggle().findNativeInput().click();
+      expectVisibilityStatus({ wrapper: contentDisplayPreferenceWrapper, statuses: [true, true, false, true] });
+    });
+
+    it('forces non-editable options to be visible', () => {
+      const wrapper = renderContentDisplay();
+      const toggleInput = wrapper.findOptions()[0].findToggle().findNativeInput();
+      expect(toggleInput.getElement()).toBeChecked();
+      expect(toggleInput.getElement()).toBeDisabled();
+      toggleInput.click();
+      expect(toggleInput.getElement()).toBeChecked();
+      expect(toggleInput.getElement()).toBeDisabled();
+    });
+  });
+
+  describe('Content reordering', () => {
+    it('moves item down', async () => {
+      const wrapper = renderContentDisplay();
+      testOrder({ wrapper, order: [0, 1, 2, 3] });
+      const dragHandle = wrapper.findOptions()[0].findDragHandle().getElement();
+      pressKey(dragHandle, 'Space');
+      await expectAnnouncement(wrapper, 'Picked up item at position 1 of 4');
```

Comment:
Thanks! Unfortunately it didn't work for me, it doesn't seem to track text changes in the same node. (Also note, this would apply to integration tests but this file in particular is a unit test. I am testing live announcements in both)